engine/src/main/java/com/datatorrent/stram/util/LoggerUtil.java

```diff
@@ -464,4 +474,55 @@ public static boolean removeAppender(String name)
     }
     return names;
   }
+
+  /**
+   * Makes MDC properties
+   */
+  public static void setupMDC()
+  {
+    String node = StramClientUtils.getHostName();
+    MDC.put("node", node == null ? "unknown" : node);
+
+    String value = System.getenv("HADOOP_USER_NAME");
+    if (value != null) {
+      MDC.put("user", value);
+    }
+
+    value = System.getenv("CONTAINER_ID");
+    if (value != null) {
+      MDC.put("containerId", value);
+    }
+
+    value = System.getProperties().getProperty(APPLICATION_NAME.getLongName());
+    if (value != null) {
+      MDC.put("application", value);
+    }
+
+    if (MDC.get("service") == null) {
+      MDC.put("service", getService());
+    }
+  }
+
+  /**
+   * Returns the service name
+   * @return Service name
+   */
+  private static String getService()
```

Comment:
I am not convinced that getting service name implicitly is required. There are just two or three places where an explicit call to setupMDC() will be added, in case it is explicit and it will help in case later somebody needs to understand where and how service names are set. The implicit setup makes it much harder to track. I would agree with the implicit initialization in case it would be necessary to insert setupMDC() in multiple locations.